airbyte-cdk/python/airbyte_cdk/sources/streams/http/http_client.py

```diff
@@ -136,34 +147,92 @@ def _create_prepared_request(
 
         return prepared_request
 
-    def _send_with_retry(self, request: requests.PreparedRequest, request_kwargs: Mapping[str, Any]) -> requests.Response:
+    def _max_retries(self) -> int:
```

Comment:
Should we consider adding the `@cached_property` decorator to these methods?

Reply:
Good idea! Added.

Reply:
That assumes that `self._error_handler.max_retries` is a constant in our current implementations. I think it is but I just wanted to point that out

Reply:
this assumes all implementations of `max_retries` are constant. Since ErrorHandler is an interface that can be customized by any connector, I don't think we can make that assumption